my-app/src/components/GenericForm.tsx

```diff
@@ -0,0 +1,147 @@
+import React, { useState } from "react";
+import { useForm } from "react-hook-form";
+import { Box } from "@mui/material";
+import FormHelperText from "@mui/material/FormHelperText";
+import Ajv, { JSONSchemaType, Schema } from "ajv";
+import ajvErrors from "ajv-errors";
+
+const ajv = new Ajv({ allErrors: true });
+require("ajv-errors")(ajv);
+
+interface Field {
+  id: string;
+  label: string;
+  type: string;
+  required: boolean;
+  placeholder: string;
+  errorMsg: string;
+  errorMsgVisibility: boolean;
+  options?: { value: string; label: string }[]; // For select fields
+}
+
+interface Props {
+  fields: Field[];
+  customSubmitFunction: (data: Record<string, any>) => void;
+  submitButtonName: string;
+  schema: Schema;
+}
+
+const GenericForm: React.FC<Props> = ({
+  fields,
+  customSubmitFunction,
+  submitButtonName,
+  schema,
+}) => {
+  const [formData, setFormData] = useState<Record<string, string | undefined>>(
+    {}
+  );
+  const [fieldsData, setFieldsData] = useState(fields);
+  const [currentLabelStyle, setCurrentLabelStyle] =
+    useState("signinLabelNormal");
+  const validate = ajv.compile(schema);
+
+  const {
+    register,
+    handleSubmit,
+    setValue,
+    formState: { errors },
+  } = useForm();
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
```

Comment:
You dont need to custom write handleInputChange . React hook form should handle this for you please dive in more in the docs .